packages/gitbook/src/lib/visitors.ts

```diff
@@ -326,16 +326,21 @@ export function getVisitorAuthCookieValue(basePath: string, token: string): stri
 }
 
 /**
- * Normalize the URL by removing the visitor authentication token from the query parameters (if present).
+ * Normalize the URL by removing the visitor JWT token and visitor.* param from the query parameters (if present).
  */
-export function normalizeVisitorAuthURL(url: URL): URL {
+export function normalizeVisitorParamsURL(url: URL): URL {
+    const withVisitorParamsURL = new URL(url);
     if (url.searchParams.has(VISITOR_AUTH_PARAM)) {
-        const withoutVAParam = new URL(url);
-        withoutVAParam.searchParams.delete(VISITOR_AUTH_PARAM);
-        return withoutVAParam;
+        withVisitorParamsURL.searchParams.delete(VISITOR_AUTH_PARAM);
     }
 
-    return url;
+    for (const [urlParam] of url.searchParams.entries()) {
+        if (urlParam.startsWith('visitor.')) {
```

Comment:
I think we use `visitor.` in a bunch of other places too. Maybe we can extract it to a `VISITOR_PARAM_PREFIX`